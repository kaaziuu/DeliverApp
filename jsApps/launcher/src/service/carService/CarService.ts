import axios, { AxiosError, AxiosResponse } from "axios";
import { useMutation } from "react-query";
import { HandleApiError } from "service/_core/HandleApiError";
import { BaseResponse, MutationProcessing } from "service/_core/Models";
import { UseStore } from "stores/Store";
import Endpoints from "utils/axios/Endpoints";
import GetHeader from "utils/axios/GetHeader";
import Config from "utils/_core/Config";
import AssignCarToDeliveryForm from "./models/AssignCarToDeliveryForm";
import AssingUserToCarForm from "./models/AssingUserToCarForm";
import CreateCarForm from "./models/CreateCarForm";
import UpdateCarForm from "./models/UpdateCarForm";

const CreateCarRequest = async (model: CreateCarForm, jwt: string): Promise<BaseResponse> => {
    const response = axios
        .post<CreateCarForm, AxiosResponse<BaseResponse>>(`${Config.serverUrl}${Endpoints.Car.Create}`, model, {
            headers: GetHeader(jwt),
        })
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError(error));

    return response;
};

export const CreateCarAction = (): MutationProcessing<CreateCarForm, BaseResponse> => {
    const { userStore } = UseStore();

    const { isLoading, mutateAsync } = useMutation((model: CreateCarForm) =>
        CreateCarRequest(model, userStore.getUser?.jwt ?? "")
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};

const UpdateCarRequest = async (model: UpdateCarForm, jwt: string): Promise<BaseResponse> => {
    const response = axios
        .put<UpdateCarForm, AxiosResponse<BaseResponse>>(`${Config.serverUrl}${Endpoints.Car.Update}`, model, {
            headers: GetHeader(jwt),
        })
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError(error));

    return response;
};

export const UpdateCarAction = (): MutationProcessing<UpdateCarForm, BaseResponse> => {
    const { userStore } = UseStore();

    const { isLoading, mutateAsync } = useMutation((form: UpdateCarForm) =>
        UpdateCarRequest(form, userStore.getUser?.jwt ?? "")
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};

const AssingUserToCarRequest = async (model: AssingUserToCarForm, jwt: string): Promise<BaseResponse> => {
    const response = await axios
        .put<AssingUserToCarForm, AxiosResponse<BaseResponse>>(
            `${Config.serverUrl}${Endpoints.Car.AssingUserToCar}`,
            model,
            { headers: GetHeader(jwt) }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError(error));

    return response;
};

export const AssingUserToCarAction = (): MutationProcessing<AssingUserToCarForm, BaseResponse> => {
    const { userStore } = UseStore();

    const { isLoading, mutateAsync } = useMutation((form: AssingUserToCarForm) =>
        AssingUserToCarRequest(form, userStore.getUser?.jwt ?? "")
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};

const AssingCarToDeliveryRequest = async (model: AssignCarToDeliveryForm, jwt: string): Promise<BaseResponse> => {
    const response = await axios
        .put<AssignCarToDeliveryForm, AxiosResponse<BaseResponse>>(
            `${Config.serverUrl}${Endpoints.Car.AssingCarToDelivery}`,
            model,
            { headers: GetHeader(jwt) }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError(error));

    return response;
};

export const AssingCarToDeliveryAction = (): MutationProcessing<AssignCarToDeliveryForm, BaseResponse> => {
    const { userStore } = UseStore();

    const { isLoading, mutateAsync } = useMutation((form: AssignCarToDeliveryForm) =>
        AssingCarToDeliveryRequest(form, userStore.getUser?.jwt ?? "")
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};
