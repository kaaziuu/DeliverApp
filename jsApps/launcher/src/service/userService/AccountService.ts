import axios, { AxiosError, AxiosResponse } from "axios";
import { useMutation, useQuery } from "react-query";
import { HandleApiError } from "service/_core/HandleApiError";
import { BaseResponse, FetchProcessing, MutationProcessing } from "service/_core/Models";
import { UseStore } from "stores/Store";
import Endpoints from "utils/axios/Endpoints";
import GetHeader from "utils/axios/GetHeader";
import Config from "utils/_core/Config";
import ChangePasswordForm from "./models/AccountModels/ChangePasswordForm";

const ChangePasswordRequest = async (request: ChangePasswordForm, jwt: string): Promise<BaseResponse<null>> => {
    const header = GetHeader(jwt);

    const response = await axios
        .put<ChangePasswordForm, AxiosResponse<BaseResponse<null>>>(
            `${Config.serverUrl}${Endpoints.Account.ChagnePassword}`,
            request,
            { headers: header }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError(error));

    return response;
};

export const ChangePasswordAction = (): MutationProcessing<ChangePasswordForm, BaseResponse<null>> => {
    const { userStore } = UseStore();

    const { isLoading, data, mutateAsync } = useMutation((form: ChangePasswordForm) =>
        ChangePasswordRequest(form, userStore.getUser!.jwt)
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
        data: data,
        error: data?.error,
        isSuccess: data?.isSuccess,
    };
};

const IsValidRecoveryKeyRequest = async (key: string): Promise<BaseResponse<Boolean>> => {
    const reponse = await axios
        .get<null, AxiosResponse<BaseResponse<Boolean>>>(
            `${Config.serverUrl}${Endpoints.Account.PasswordRecoverySetNewPassword}`
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError<Boolean>(error));

    return reponse;
};

export const IsValidRecoveryKeyAction = (recoveryKey: string): FetchProcessing<Boolean> => {
    const { isLoading, data, refetch } = useQuery("IsValidRecoveryKeyRequest", () =>
        IsValidRecoveryKeyRequest(recoveryKey)
    );
    return {
        isLoading: isLoading,
        data: data?.data,
        isSuccess: data?.isSuccess,
        error: data?.error,
        refresh: refetch,
    };
};
