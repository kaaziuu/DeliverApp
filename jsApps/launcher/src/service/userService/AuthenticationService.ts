import axios, { AxiosError, AxiosResponse } from "axios";
import { useMutation, useQuery } from "react-query";
import { useNavigate } from "react-router-dom";
import GetHeader from "utils/axios/GetHeader";
import Path from "utils/route/Path";
import { UseStore } from "../../stores/Store";
import Endpoints from "../../utils/axios/Endpoints";
import Config from "../../utils/_core/Config";
import { BaseResponse, FetchProcessing, MutationProcessing } from "../_core/Models";
import AuthResponse from "./models/AuthResponse";
import LoginForm from "./models/LoginForm";

export const LoginRequest = async (form: LoginForm): Promise<BaseResponse<AuthResponse>> => {
    const response = await axios
        .post<LoginForm, AxiosResponse<BaseResponse<AuthResponse>>>(
            `${Config.serverUrl}${Endpoints.Authentication.Login}`,
            form,
            {
                withCredentials: true,
            }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => {
            return {
                isSuccess: false,
                error: error.message,
            } as BaseResponse<AuthResponse>;
        });
    return response;
};

const RefreshTokenRequest = async (): Promise<BaseResponse<AuthResponse>> => {
    const response = await axios
        .post<undefined, AxiosResponse<BaseResponse<AuthResponse>>>(
            `${Config.serverUrl}${Endpoints.Authentication.Refresh}`,
            null,
            {
                withCredentials: true,
            }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => {
            return {
                isSuccess: false,
                error: error.message,
            } as BaseResponse<AuthResponse>;
        });
    return response;
};

export const RefreshToken = (): FetchProcessing<AuthResponse> => {
    const { userStore } = UseStore();
    const { isLoading, data } = useQuery("RefreshToken", RefreshTokenRequest, {
        refetchInterval: 180000,
        onError: () => {
            userStore.logout();
        },
        onSuccess: (response) => {
            if (!response.isSuccess) {
                userStore.logout();
            }
        },
    });

    if (!isLoading && data?.isSuccess) {
        userStore.setUser(data!.data!);
    }

    return {
        isLoading: isLoading,
        error: data?.error,
        isSuccess: data?.isSuccess,
        data: data?.data,
    };
};

export const Login = (): MutationProcessing<LoginForm, BaseResponse<AuthResponse>> => {
    const { userStore } = UseStore();
    const navigation = useNavigate();

    const { isLoading, data, mutate, mutateAsync } = useMutation((request: LoginForm) => LoginRequest(request), {
        onSuccess: (result) => {
            if (result?.isSuccess) {
                userStore.setUser(result!.data!);
            } else {
                navigation(Path.login);
            }
        },
    });

    return {
        isLoading: isLoading,
        error: data?.error,
        isSuccess: data?.isSuccess,
        data: data,
        mutateAsync: mutateAsync,
    };
};

const LogoutRequest = async (jwt: string): Promise<BaseResponse<null>> => {
    const reponse = await axios
        .post<null, AxiosResponse<BaseResponse<null>>>(
            `${Config.serverUrl}${Endpoints.Authentication.Logout}`,
            {},
            { headers: GetHeader(jwt) }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => ({ error: error.message, isSuccess: false } as BaseResponse<null>));

    return reponse;
};

export const Logout = (): MutationProcessing<undefined, BaseResponse<null>> => {
    const { userStore } = UseStore();

    const { isLoading, data, mutate, mutateAsync } = useMutation("logout", () => LogoutRequest(userStore.getUser!.jwt));

    return {
        isLoading: isLoading,
        data: data,
        error: data?.error,
        isSuccess: data?.isSuccess,
        mutateAsync: mutateAsync,
    };
};
