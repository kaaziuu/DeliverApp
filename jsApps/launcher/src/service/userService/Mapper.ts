import AuthResponse from "./models/AuthModels/AuthResponse";
import User from "./models/User";

export const MapAuthResponseToUser = (response: AuthResponse): User => {
    return {
        hash: response.hash,
        jwt: response.jwt,
        name: response.name,
        roles: response.roles,
        surname: response.surname,
        username: response.surname,
        expireDate: response.expireDate,
        companyHash: response.companyHash,
    };
};

export default MapAuthResponseToUser;
