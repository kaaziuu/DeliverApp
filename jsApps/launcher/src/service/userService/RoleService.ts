import axios, { AxiosError, AxiosRequestHeaders, AxiosResponse } from "axios";
import { useQuery } from "react-query";
import { BaseResponse, FetchProcessing } from "service/_core/Models";
import { UseStore } from "stores/Store";
import Endpoints from "utils/axios/Endpoints";
import GetHeader from "utils/axios/GetHeader";
import Config from "utils/_core/Config";
import RoleResponse from "./models/RoleResponse";

const FetchRole = async (header: AxiosRequestHeaders): Promise<BaseResponse<RoleResponse[]>> => {
    const response = await axios
        .get<null, AxiosResponse<BaseResponse<RoleResponse[]>>>(`${Config.serverUrl}${Endpoints.Role.GetAll}`, {
            headers: header,
        })
        .then((resp) => resp.data)
        .catch(
            (error: AxiosError) =>
                ({
                    isSuccess: false,
                    error: error.message,
                } as BaseResponse<RoleResponse[]>)
        );

    return response;
};

export const GetRoles = (): FetchProcessing<RoleResponse[]> => {
    const { userStore } = UseStore();
    const header = GetHeader(userStore.getUser!.jwt);
    const { data, isLoading } = useQuery("fetch roles", () => FetchRole(header), { cacheTime: Infinity });

    return {
        isLoading: isLoading,
        data: data?.data,
        error: data?.error,
        isSuccess: data?.isSuccess,
    };
};
