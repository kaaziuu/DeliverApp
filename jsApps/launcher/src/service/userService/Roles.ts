export const Roles = {
    Admin: "Admin",
    CompanyAdmin: "CompanyAdmin",
    CompanyOwner: "CompanyOwner",
    Hr: "HR",
    Driver: "Driver",
    Dispatcher: "Dispatcher",
};

const HasRole = (roles: string[], requiresRole: string[]): boolean => {
    const hasRole = roles.some((x) => requiresRole.some((y) => y === x));
    return hasRole;
};

export default HasRole;
