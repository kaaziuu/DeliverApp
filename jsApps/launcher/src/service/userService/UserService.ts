import axios, { AxiosError, AxiosRequestHeaders, AxiosResponse } from "axios";
import { useMutation, useQuery } from "react-query";
import { HandleApiError } from "service/_core/HandleApiError";
import { UseStore } from "stores/Store";
import Endpoints from "utils/axios/Endpoints";
import GetHeader from "utils/axios/GetHeader";
import Config from "utils/_core/Config";
import { BaseResponse, FetchProcessing, MutationProcessing } from "../_core/Models";
import CreateUserForm from "./models/UserModels/CreateUserForm";
import UpdateUserForm from "./models/UserModels/UpdateUserForm";
import UserResponse from "./models/UserModels/UserResponse";

const CreateUserRequest = async (
    form: CreateUserForm,
    header: AxiosRequestHeaders
): Promise<BaseResponse<UserResponse>> => {
    const response = await axios
        .post<CreateUserForm, AxiosResponse<BaseResponse<UserResponse>>>(
            `${Config.serverUrl}${Endpoints.User.Create}`,
            form,
            { headers: header }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError<UserResponse>(error));
    return response;
};

export const CreateUser = (): MutationProcessing<CreateUserForm, BaseResponse<UserResponse>> => {
    const { userStore } = UseStore();
    const header = GetHeader(userStore.getUser!.jwt);

    const { isLoading, mutateAsync } = useMutation((form: CreateUserForm) => CreateUserRequest(form, header), {
        onMutate: (form) => {
            form.companyHash = userStore.getUser!.companyHash;
        },
    });

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};

const GetUserRequest = async (hash: string, jwt: string): Promise<BaseResponse<UserResponse>> => {
    const header = GetHeader(jwt);

    const response = await axios
        .get<null, AxiosResponse<BaseResponse<UserResponse>>>(`${Config.serverUrl}${Endpoints.User.GetUser(hash)}`, {
            headers: header,
        })
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError<UserResponse>(error));

    return response;
};

export const GetUser = (
    hash: string,
    onSuccess?: (result: BaseResponse<UserResponse>) => void
): FetchProcessing<UserResponse> => {
    const { userStore } = UseStore();

    const { isLoading, data } = useQuery(["GetUser", hash], () => GetUserRequest(hash, userStore.getUser!.jwt), {
        cacheTime: Infinity,
        onSuccess: onSuccess,
    });

    return {
        isLoading: isLoading,
        isSuccess: data?.isSuccess,
        data: data?.data,
        error: data?.error,
    };
};

const UpdateUserRequest = async (request: UpdateUserForm, jwt: string): Promise<BaseResponse<UserResponse>> => {
    const header = GetHeader(jwt);

    const response = await axios
        .put<UpdateUserForm, AxiosResponse<BaseResponse<UserResponse>>>(
            `${Config.serverUrl}${Endpoints.User.UpdateUser}`,
            request,
            {
                headers: header,
            }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError<UserResponse>(error));

    return response;
};

export const UpdateUserAction = (): MutationProcessing<UpdateUserForm, BaseResponse<UserResponse>> => {
    const { userStore } = UseStore();

    const { isLoading, mutateAsync } = useMutation("update user", (form: UpdateUserForm) =>
        UpdateUserRequest(form, userStore.getUser!.jwt)
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};

const FetchWorkers = async (header: AxiosRequestHeaders): Promise<BaseResponse<UserResponse[]>> => {
    const response = await axios
        .get<BaseResponse<UserResponse[]>>(`${Config.serverUrl}${Endpoints.User.UserList}`, {
            headers: header,
        })
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError<UserResponse[]>(error));
    return response;
};

export const GetWorkers = (): FetchProcessing<UserResponse[]> => {
    const { userStore } = UseStore();

    const header = GetHeader(userStore.getUser!.jwt);
    const { data, isLoading, refetch } = useQuery("fetch workers", () => FetchWorkers(header), { cacheTime: Infinity });

    return {
        isLoading: isLoading,
        data: data?.data,
        error: data?.error,
        isSuccess: data?.isSuccess,
        refresh: refetch,
    };
};

const FireUserRequest = async (userHash: string, jwt: string): Promise<BaseResponse<null>> => {
    const response = await axios
        .put<null, AxiosResponse<BaseResponse<null>>>(
            `${Config.serverUrl}${Endpoints.User.Fire(userHash)}`,
            undefined,
            { headers: GetHeader(jwt) }
        )
        .then((resp) => resp.data)
        .catch((error: AxiosError) => HandleApiError(error));

    return response;
};

export const FireUserAction = (): MutationProcessing<string, BaseResponse<null>> => {
    const { userStore } = UseStore();

    const { isLoading, mutateAsync } = useMutation((hash: string) =>
        FireUserRequest(hash, userStore.getUser?.jwt ?? "")
    );

    return {
        isLoading: isLoading,
        mutateAsync: mutateAsync,
    };
};
