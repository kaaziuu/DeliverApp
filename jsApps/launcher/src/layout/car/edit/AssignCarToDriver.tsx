import LoadingButton from "@mui/lab/LoadingButton";
import { CircularProgress, InputLabel, MenuItem, Select } from "@mui/material";
import Snackbar from "components/snackbar/Snackbar";
import { useState } from "react";
import { AssingUserToCarAction } from "service/carService/CarService";
import AssingUserToCarForm, { GetDefaultAssingUserToCarForm } from "service/carService/models/AssingUserToCarForm";
import { GetDriverAction } from "service/userService/UserService";
import CreateClass from "utils/style/CreateClass";

interface props {
    carHash: string;
    driverHash?: string;
}

const baseClass = "car-edit_assign-car-to-driver";

const AssignCarToDriver = ({ carHash, driverHash }: props) => {
    const [form, setForm] = useState<AssingUserToCarForm>(GetDefaultAssingUserToCarForm(carHash, driverHash));
    const [error, setError] = useState<string | null>(null);
    const [showSnackbar, setShowSnackbar] = useState(false);
    const drivers = GetDriverAction();
    const assingUserToCarAction = AssingUserToCarAction();

    const submit = async () => {
        const result = await assingUserToCarAction.mutateAsync(form);
        setError(result.error);
        setShowSnackbar(true);
    };

    if (!drivers.isSuccess) {
        return null;
    }

    return (
        <>
            <div className={baseClass}>
                {drivers.isLoading ? (
                    <CircularProgress />
                ) : (
                    <>
                        <div className={CreateClass(baseClass, "select")}>
                            <InputLabel>Select driver: </InputLabel>
                            <Select
                                id="select-driver"
                                value={form.userHash}
                                onChange={(e) => setForm({ ...form, userHash: e.target.value })}
                            >
                                {drivers.data!.map((driver) => (
                                    <MenuItem
                                        value={driver.hash}
                                        key={driver.hash}
                                    >{`${driver.name} ${driver.surname}`}</MenuItem>
                                ))}
                            </Select>
                        </div>
                        <LoadingButton variant="contained" loading={assingUserToCarAction.isLoading} onClick={submit}>
                            Assing
                        </LoadingButton>
                    </>
                )}
            </div>
            <Snackbar
                message={error ? error : "User assing to car"}
                setIsOpen={setShowSnackbar}
                variant={error ? "error" : "success"}
                isOpen={showSnackbar}
            />
        </>
    );
};

export default AssignCarToDriver;
