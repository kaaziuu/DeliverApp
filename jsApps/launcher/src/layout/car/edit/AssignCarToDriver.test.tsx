import { render } from "@testing-library/react";
import { AssingUserToCarAction } from "service/carService/CarService";
import { GetDriverAction } from "service/userService/UserService";
import AssignCarToDriver from "./AssignCarToDriver";

jest.mock("service/userService/UserService", () => ({
    GetDriverAction: jest.fn(),
}));

jest.mock("service/carService/CarService", () => ({
    AssingUserToCarAction: jest.fn(),
}));

describe(AssignCarToDriver, () => {
    test("should match to snapshot", () => {
        (GetDriverAction as jest.MockedFunction<typeof GetDriverAction>).mockReturnValue({
            isLoading: false,
            data: [
                {
                    hash: "test",
                    companyHash: "",
                    companyName: "",
                    email: "",
                    name: "",
                    roles: [],
                    surname: "",
                    username: "",
                },
            ],
            isSuccess: true,
        });

        (AssingUserToCarAction as jest.MockedFunction<typeof AssingUserToCarAction>).mockReturnValue({
            isLoading: false,
            mutateAsync: jest.fn(),
        });

        const compoent = render(<AssignCarToDriver carHash="asd" />);

        expect(compoent).toMatchSnapshot();
    });
});
