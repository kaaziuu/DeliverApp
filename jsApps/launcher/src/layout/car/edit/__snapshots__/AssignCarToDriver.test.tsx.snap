// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignCarToDriver should match to snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="car-edit_assign-car-to-driver"
      >
        <div
          class="car-edit_assign-car-to-driver-select"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
          >
            Select driver: 
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select-driver"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="select-driver"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-hdw1oc"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          id="mui-1"
          tabindex="0"
          type="button"
        >
          Assing
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="car-edit_assign-car-to-driver"
    >
      <div
        class="car-edit_assign-car-to-driver-select"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
        >
          Select driver: 
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="select-driver"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="select-driver"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
        id="mui-1"
        tabindex="0"
        type="button"
      >
        Assing
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
